example/contrib/socks-client/socks/impl/error.ipp

```diff
@@ -0,0 +1,97 @@
+//
+// Copyright (c) 2018 jackarain (jack dot wgm at gmail dot com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/beast
+//
+
+#ifndef SOCKS_IMPL_ERROR_IPP
+#define SOCKS_IMPL_ERROR_IPP
+
+#include <socks/error.hpp>
+
+namespace socks {
+
+namespace detail {
+
+class error_codes : public error_category
```

Comment:
There's a better way to structure this code:
https://github.com/vinniefalco/json/blob/develop/include/boost/json/impl/error.ipp#L19
You tuck it all into the function. It isn't urgent to change this now but FYI.

Reply:
so done.